[Java] JDK11 string jit serializaiton failure

Thank you for reporting the problem!
Please make sure what you are reporting is a bug with reproducible steps. 

**Describe the bug**
#531 will make jdk11 string jit serialization failed with:

```
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:131)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:69)
	at io.fury.codegen.CodeGenerator.compile(CodeGenerator.java:143)
	at io.fury.builder.CodecUtils.loadOrGenCodecClass(CodecUtils.java:91)
	at io.fury.builder.CodecUtils.loadOrGenObjectCodecClass(CodecUtils.java:42)
	at io.fury.serializer.CodegenSerializer.loadCodegenSerializer(CodegenSerializer.java:58)
	... 23 more
Caused by: org.codehaus.commons.compiler.CompileException: File 'io/fury/benchmark/data/SampleFuryCodec_2085539722_125130493_341546948.java', Line 186, Column 42: No applicable constructor/method found for actual parameters "io.fury.memory.MemoryBuffer, java.lang.String"; candidates are: "default static void io.fury.serializer.StringSerializer.writeJDK11String(io.fury.memory.MemoryBuffer, java.lang.String)"
	at org.codehaus.janino.UnitCompiler.compileError(UnitCompiler.java:12211)
	at org.codehaus.janino.UnitCompiler.findMostSpecificIInvocable(UnitCompiler.java:9263)
	at org.codehaus.janino.UnitCompiler.findIMethod(UnitCompiler.java:9123)
	at org.codehaus.janino.UnitCompiler.findIMethod(UnitCompiler.java:9025)
	at org.codehaus.janino.UnitCompiler.compileGet2(UnitCompiler.java:5062)
	at org.codehaus.janino.UnitCompiler.access$9100(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$16.visitMethodInvocation(UnitCompiler.java:4423)
	at org.codehaus.janino.UnitCompiler$16.visitMethodInvocation(UnitCompiler.java:4396)
	at org.codehaus.janino.Java$MethodInvocation.accept(Java.java:5073)
	at org.codehaus.janino.UnitCompiler.compileGet(UnitCompiler.java:4396)
	at org.codehaus.janino.UnitCompiler.compileGetValue(UnitCompiler.java:5662)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:3783)
	at org.codehaus.janino.UnitCompiler.access$5900(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$13.visitMethodInvocation(UnitCompiler.java:3762)
	at org.codehaus.janino.UnitCompiler$13.visitMethodInvocation(UnitCompiler.java:3734)
	at org.codehaus.janino.Java$MethodInvocation.accept(Java.java:5073)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:3734)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2360)
	at org.codehaus.janino.UnitCompiler.access$1800(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$6.visitExpressionStatement(UnitCompiler.java:1494)
	at org.codehaus.janino.UnitCompiler$6.visitExpressionStatement(UnitCompiler.java:1487)
	at org.codehaus.janino.Java$ExpressionStatement.accept(Java.java:2874)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1487)
	at org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1567)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:1553)
	at org.codehaus.janino.UnitCompiler.access$1700(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1493)
	at org.codehaus.janino.UnitCompiler$6.visitBlock(UnitCompiler.java:1487)
	at org.codehaus.janino.Java$Block.accept(Java.java:2779)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1487)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2468)
	at org.codehaus.janino.UnitCompiler.access$1900(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1495)
	at org.codehaus.janino.UnitCompiler$6.visitIfStatement(UnitCompiler.java:1487)
	at org.codehaus.janino.Java$IfStatement.accept(Java.java:2950)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1487)
	at org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1567)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:3388)
	at org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1357)
	at org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1330)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:822)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:432)
	at org.codehaus.janino.UnitCompiler.access$400(UnitCompiler.java:215)
	at org.codehaus.janino.UnitCompiler$2.visitPackageMemberClassDeclaration(UnitCompiler.java:411)
	at org.codehaus.janino.UnitCompiler$2.visitPackageMemberClassDeclaration(UnitCompiler.java:406)
	at org.codehaus.janino.Java$PackageMemberClassDeclaration.accept(Java.java:1414)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:406)
	at org.codehaus.janino.UnitCompiler.compileUnit(UnitCompiler.java:378)
	at org.codehaus.janino.Compiler.compile(Compiler.java:532)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:111)
	... 28 more
```

We need to make `io.fury.serializer.StringSerializer#writeJDK11String` public 
**Expected behavior**


**To Reproduce**
Steps to reproduce the behavior.
Please try to provide a reproducible script.

**Screenshots**
![image](https://github.com/alipay/fury/assets/12445254/d853555e-1630-4f20-84c9-087d3b665f16)
**Environment (please complete the following information):**
Please specify the versions of Fury, OS, and others such as JDK/Python/Gcc/Clang/Go/NodeJS if they are used.

--- a/java/fury-benchmark/src/main/java/io/fury/benchmark/NewJava11StringSuite.java
+++ b/java/fury-benchmark/src/main/java/io/fury/benchmark/NewJava11StringSuite.java
@@ -70,7 +70,7 @@ public class NewJava11StringSuite {
 
   // @Benchmark
   public Object createJDK8StringByMethodHandle() {
-    return StringSerializer.newJava11StringByZeroCopy(strBytes, coder);
+    return StringSerializer.newJava11StringByZeroCopy(coder, strBytes);
   }
 
   // @Benchmark

--- a/java/fury-core/src/main/java/io/fury/serializer/StringSerializer.java
+++ b/java/fury-core/src/main/java/io/fury/serializer/StringSerializer.java
@@ -137,7 +137,7 @@ public final class StringSerializer extends Serializer<String> {
   public Expression writeStringExpr(Expression strSerializer, Expression buffer, Expression str) {
     if (isJava) {
       if (STRING_VALUE_FIELD_IS_BYTES) {
-        return new Invoke(strSerializer, "writeJDK11String", buffer, str);
+        return new StaticInvoke(StringSerializer.class, "writeJDK11String", buffer, str);
       } else {
         if (!STRING_VALUE_FIELD_IS_CHARS) {
           throw new UnsupportedOperationException();
@@ -188,7 +188,7 @@ public final class StringSerializer extends Serializer<String> {
           Expression coder = inlineInvoke(buffer, "readByte", BYTE_TYPE);
           Expression value = inlineInvoke(buffer, "readBytesWithSizeEmbedded", BINARY_TYPE);
           return new StaticInvoke(
-              StringSerializer.class, "newJava11StringByZeroCopy", STRING_TYPE, value, coder);
+              StringSerializer.class, "newJava11StringByZeroCopy", STRING_TYPE, coder, value);
         }
       } else {
         if (!STRING_VALUE_FIELD_IS_CHARS) {
@@ -298,7 +298,7 @@ public final class StringSerializer extends Serializer<String> {
       } else {
         byte coder = buffer.readByte();
         byte[] value = buffer.readBytesWithSizeEmbedded();
-        return newJava11StringByZeroCopy(value, coder);
+        return newJava11StringByZeroCopy(coder, value);
       }
     } else {
       if (!STRING_VALUE_FIELD_IS_CHARS) {
@@ -317,7 +317,7 @@ public final class StringSerializer extends Serializer<String> {
     }
   }
 
-  static void writeJDK11String(MemoryBuffer buffer, String value) {
+  public static void writeJDK11String(MemoryBuffer buffer, String value) {
     byte[] bytes = (byte[]) Platform.getObject(value, STRING_VALUE_FIELD_OFFSET);
     byte coder = Platform.getByte(value, STRING_CODER_FIELD_OFFSET);
     buffer.writeByte(coder);
@@ -472,7 +472,9 @@ public final class StringSerializer extends Serializer<String> {
     }
   }
 
-  public static String newJava11StringByZeroCopy(byte[] data, byte coder) {
+  // coder param first to make inline call args
+  // `(buffer.readByte(), buffer.readBytesWithSizeEmbedded())` work.
+  public static String newJava11StringByZeroCopy(byte coder, byte[] data) {
     if (Platform.JAVA_VERSION < 9) {
       throw new IllegalStateException(
           String.format("Current java version is %s", Platform.JAVA_VERSION));

--- a/java/fury-core/src/test/java/io/fury/serializer/StringSerializerTest.java
+++ b/java/fury-core/src/test/java/io/fury/serializer/StringSerializerTest.java
@@ -92,7 +92,7 @@ public class StringSerializerTest extends FuryTestBase {
   static String readJDK11String(MemoryBuffer buffer) {
     byte coder = buffer.readByte();
     byte[] value = buffer.readBytesWithSizeEmbedded();
-    return newJava11StringByZeroCopy(value, coder);
+    return newJava11StringByZeroCopy(coder, value);
   }
 
   private static boolean writeJavaStringZeroCopy(MemoryBuffer buffer, String value) {

